common4j/build.gradle

```diff
@@ -24,6 +25,106 @@ repositories {
+
+version = getAppVersionName()
+
+task sourcesJar(type: Jar) {
+    from sourceSets.main.java.srcDirs
+    classifier = 'sources'
+    destinationDirectory = reporting.file("$project.buildDir/libs/")
+}
+
+// Task to generate javadoc
+task generateJavadoc(type: Javadoc) {
+    failOnError false
+    title = "Microsoft Identity Common Java"
+    source = sourceSets.main.java
+    classpath += project.sourceSets.main.compileClasspath
+
+    options.memberLevel = JavadocMemberLevel.PUBLIC
+    options.addStringOption('Xdoclint:none', '-quiet')
+
+    exclude '**/BuildConfig.Java'
+    exclude '**/R.java'
+    destinationDir = reporting.file("$project.buildDir/javadoc")
+}
+
+// Task to generate javadoc.jar
+task javadocJar(type: Jar, dependsOn: generateJavadoc) {
+    from javadoc.destinationDir
+    classifier 'javadoc'
+    destinationDirectory = reporting.file("$project.buildDir/libs")
+}
+
+jar {
+    manifest {
+        attributes('Implementation-Title': project.name,
+                'Implementation-Version': project.version)
+    }
+}
+
+publishing {
+    publications {
+        aar(MavenPublication) {
+            groupId 'com.microsoft.identity'
+            artifactId 'common4j'
+            from components.java
+
+            pom {
+                name = 'common4j'
+                description = 'This library contains common java code shared between the Active Directory ' +
+                        'Authentication Library (ADAL) for Android and the Microsoft ' +
+                        'Authentication Library (MSAL) for Android. This library ' +
+                        'includes only internal classes and is NOT part of the ' +
+                        'public API'
+                url = 'https://github.com/AzureAD/microsoft-authentication-library-common-for-android'
+                developers {
+                    developer {
+                        id = 'microsoft'
+                        name = 'Microsoft'
+                    }
+                }
+                licenses {
+                    license {
+                        name = 'MIT License'
+                    }
+                }
+                inceptionYear = '2021'
+                scm {
+                    url = 'https://github.com/AzureAD/microsoft-authentication-library-common-for-android/tree/master'
+                }
+                properties = [
+                        branch: 'master',
+                        version: project.version
+                ]
+            }
+        }
+    }
+
+    repositories {
+        maven {
+            name "vsts-maven-adal-android"
+            url "https://identitydivision.pkgs.visualstudio.com/_packaging/AndroidADAL/maven/v1"
+            credentials {
+                username project.ext.vstsUsername
+                password project.ext.vstsPassword
+            }
+        }
+        maven {
+            name "vsts-maven-android"
```

Comment:
Do we still use this?

Reply:
I actually don't know. I just copied it from other projects. I am going to leave this in for now to avoid any surprises and we can come back to this later.